Rework on PeerCard menu

grummbeer opened the pull request:
### Fixes

- menu item "remove" doesn't remove peer from favorites
- menu showing always "add to favorites" even peer is already a favorite
- menu item "add to addressbook" disappear when clicked

### Changes

- move menu item "remove" to the last menu position, make its destructive role visible, change text to "delete"
- add missing menu item "add/remove favorite" to DiscoveredPeersCard
- hide menu item "add to address book" if not logged in.
- add visually feedback for state of "is favorite" in menu item "add/remove favorite"
- add missing label to Text input on "rename"

---

### Menu item "remove"

The "remove" entry is too close to its siblings is some ways.

|before | after |
|-- |-- |
|![remove-before](https://user-images.githubusercontent.com/67791701/219976511-86c73c5f-3b1b-4fb6-b1ac-ba9c05f8dbb8.png)|![remove-after](https://user-images.githubusercontent.com/67791701/220903141-278e0097-b190-4e4e-be88-ac68b264b1f1.png)|

### Menu item "Add to favorites" 

"Add favorites" is always shown in menu even if already in favs. 

**Now**:

|peer not in favs | peer in favs |
|-- |-- |
|![favorite-before](https://user-images.githubusercontent.com/67791701/219976509-aaf99b7f-e0df-4f11-a3d8-217a622081ac.png)|![favorite-after](https://user-images.githubusercontent.com/67791701/219976502-8ad6f2de-d357-4639-b352-27e519e6f530.png)|


#### add visual state indicator (behave like a checkbox)

|peer not in favs | peer in favs |
|-- |-- |
|![favorite-indicator-add](https://user-images.githubusercontent.com/67791701/219979396-48c63079-c539-4433-b237-abc2db0ff521.png)|![favorite-indicator-remove](https://user-images.githubusercontent.com/67791701/219979397-8324968f-1d39-4dbd-88ae-6dd244ff6580.png)|

grummbeer committed:
PeerCard. Menu. Move "remove" to last position.

grummbeer committed:
PeerCard. Menu. "add to favorites" visual indicator

grummbeer commented:
@rustdesk @21pages  need help on this.

- "remove" does not delete peer from favs. ` fav = [ '37…']` , bug or feature?
- What is the state of the addressbook? Is it working?

rustdesk commented:
@21pages will assit you.

rustdesk commented:
> Is there a "loading" indicator for the time trying to waking up the remote machine?

No such thing, standard WOL protocol is super simple, no idea if we can get reply.

rustdesk commented:
> What is intended here?

Delete is intended to remove from current list, it does not consider to  delete from fav also.

rustdesk commented:
> ### "Add to favorites" is always shown in recent peers
> "Add favorites" is always shown in menu even if already in favs.

This is bug

rustdesk commented:
> considering overlay with [star] to make the favorite state direct visible.

Agree

rustdesk commented:
> if "Add to addressbook" clicked, the menu item is gone.

"Add to addressbook" should be invisible if not logged in.

rustdesk commented:
> * WOL too nerdisch > "Wake up remote desktop (WOL)"

Agree, To make WOL working is not easy, RustDesk only discovers the local ip and sends WOL packet out, the users need to do the other settings (BIOS etc).

21pages commented:
> "remove" does not delete peer from favs.  fav = [ '37…'] , bug or feature?

Now it will reappear in favorites after next connection.
I think when someone deletes it, they don't need it anymore.

grummbeer commented:
> "Add to addressbook" should be invisible if not logged in.

please take a look to https://github.com/rustdesk/rustdesk/pull/3283/commits/63043d41582e028be5b1ae0dbbf26ac7b1777ba4

- is `if (gFFI.userModel.userName.isNotEmpty) {` the correct way getting the logged in state?
- are the comments real todos?

--- 

user: hello, big big problem, how can i add an peer to the address book? I am logged in right now on adress book tab.
helper: have you clicked the three dot menu since that?
user: YES!! of course! several times … wait … hell, i swear five minutes ago there was no option for that.
helper: no problem, invoice to the known address?
user: WTF?!

this in mind i would prefer showing this option but make it inaccessible if not logged in.

But i don't know how. Its OK as it is now :-).

grummbeer committed:
PeerCard. Menu. Remove peer also from favorites when deleted

grummbeer committed:
PeerCard. Menu. Make "add to favorites" dynamic

grummbeer commented:
remove

> Delete is intended to remove from current list, it does not consider to delete from fav also.

good point. But currently there is no "recent peers" list from where the entry could remove. Its the raw datastore and "remove" means "delete peer completely from database"

> I think when someone deletes it, they don't need it anymore.

agree, should the button text change to "delete" instead of "remove"?

Please review commit https://github.com/rustdesk/rustdesk/pull/3283/commits/d6f2b81a283ce5b72f32584cbc1b12f32f2bb11f. it looks like a workaround and not well implemented.  Maybe its the best i drop this commit and you fix that after merge?

grummbeer commented:
DiscoveredPeerCard

has "add to Addressbook" but no "add to favorite". Intended?

rustdesk commented:
> DiscoveredPeerCard
> 
> has "add to Addressbook" but not "add to favorite". Intended?

not intended

rustdesk commented:
> remove
> 
> > Delete is intended to remove from current list, it does not consider to delete from fav also.
> 
> good point. But currently there is no "recent peers" list from where the entry could remove. Its the raw datastore and "remove" means "delete peer completely from database"
> 
> > I think when someone deletes it, they don't need it anymore.
> 
> agree, should the button text change to "delete" instead of "remove"?
> 
> Please review commit [d6f2b81](https://github.com/rustdesk/rustdesk/commit/d6f2b81a283ce5b72f32584cbc1b12f32f2bb11f). it looks like a workaround and not well implemented. Maybe its the best i drop this commit and you fix that after merge?

@21pages follow up

rustdesk commented:
@21pages review, test and feedback please.

21pages commented:
> Please review commit [d6f2b81](https://github.com/rustdesk/rustdesk/commit/d6f2b81a283ce5b72f32584cbc1b12f32f2bb11f). it looks like a workaround and not well implemented.
It's ok in my test. what's the problem?

I think you can change "Remove" to " Remove from addressbook", because "Remove" in other tab remove local database.

21pages commented:
Send me an email if convenient, pages21@163.com

grummbeer committed:
PeerCard. Menu. Add menu item "add to favorite" to DiscoveredPeerCard

grummbeer commented:
>> Please review commit [d6f2b81](https://github.com/rustdesk/rustdesk/commit/d6f2b81a283ce5b72f32584cbc1b12f32f2bb11f). it looks like a workaround and not well implemented.

> It's ok in my test. what's the problem? 

no problem.

> In addressbook menu, changing "Remove" to " Remove from addressbook" may be better …

agree, but i can not see any addressbook card. Working on a invisible widget is a bit crazy :-) 

>> DiscoveredPeerCard
>> has "add to Addressbook" but not "add to favorite". Intended?

> not intended

menu items added.

>>considering overlay with [star] to make the favorite state direct visible.
>Agree

can not implement this at the moment. I do not know how getting the peers fav state at this point.

Actually I only wanted to move the "Remove" button to the last menu position, but now it becomes too complicatet for me. i came accross that i am not ready yet doing more work on this component.

let's review and cleanup the current work and merge it, some other iterations can address all things what left here

grummbeer commented:
updated description with the state of the current work.

things I noticed but can't work on. too much for now or do not have the skills doing it.

- unclear state of addressbook, can't login, can't work on it
- unclear state of menu items "add/remove addressbook" in other lists than AddressbookCard
- what happen to a discovered peer on "remove"? Will they no longer detected? `if (isLAN) { TODO }` any todos?
- ambigous meaning of "remove".  Remove from current list (hide from view?)  or delete from database.
- card overlay for "favorite" and " addressbook" state would be nice
- looking for something like "(rx) FavoritesList" or "(rx) Peer.IsFavorite" for smoother working with its fav state
- is "WOL" too nerdisch? should the button text more common user friendly? "Wake up (WOL)" or "Start remote machine (WOL)" or similar

grummbeer committed:
PeerCard. Menu. Add label to text input on "rename"

21pages commented:
> unclear state of addressbook, can't login, can't work on it
Email me and I'll send you a server, it's under development but available

21pages commented:
BTW, would you like to design the title bar? I like the style of browsers, but just personal opinion. You can make your own design and make any changes, like titlebar height and button size.
![1677033141727](https://user-images.githubusercontent.com/14891774/220506030-feb16134-ee28-4e98-87c6-b83e03316f54.png)

grummbeer commented:
Thanks for your feedback.

>> ambigous meaning of "remove". Remove from current list (hide from view?) or delete from database.

> It's a safe choice to keep deleting the local database.

yeah, this is what currently happen. What i meant was the button text should reflect this  with "delete" instead of "remove".  To see a sample just click the three dot menu on any of your comments here in this PR. but its ok for me if you want keeping "remove"

> BTW, would you like to design the title bar? I like the style of browsers, 

See it. Think  @NicKoehler is [in flow](https://github.com/rustdesk/rustdesk/discussions/3288) maybe he can include this in mind.

21pages commented:
> this is what currently happen

Yes, "Delete" is  more explicit, it's better to use "Delete".

grummbeer committed:
PeerCard. Menu. Make "remove" more visible

grummbeer committed:
PeerCard. Menu. Change button text "remove" to "delete"

grummbeer commented:
> Yes, it actually does "Delete", so it's better to use "Delete".

Changed text to "delete"

I added at first a new named color (`static const Color danger = Colors.red;`) to the theme (binding a color to a specific purpose) but i reverted this. I think there will be a task in the futur that addresses the "create/select color on demand vs. defined colors  in theme".

Please review.

- all other what needs more "braining" can solved later. 
- If the new file transfert is fully merged,  i want to adopt the new look.

grummbeer committed:
PeerCard. Menu. Hide "Add to Addressbook" if not logged in
